packages/core/src/binary-data/binary-data.config.ts

```diff
@@ -1,8 +1,39 @@
 import { Config, Env } from '@n8n/config';
+import { createHash } from 'node:crypto';
+import path from 'node:path';
+import { z } from 'zod';
+
+import { InstanceSettings } from '@/instance-settings';
+
+const binaryDataModesSchema = z.enum(['default', 'filesystem', 's3']);
+
+const availableModesSchema = z
+	.string()
+	.transform((value) => value.split(','))
+	.pipe(binaryDataModesSchema.array());
```

Comment:
We already have `CommaSeparatedStringArray`. Do we favor Zod schemas from now on?